### ISSUE TITLE
Media comments issue: cant collect all media comments

### ISSUE THREAD

**Issue by AndManowar** [mgp25/Instagram-API#2450]
Hello, I have some problems with collecting media comments.
I cant collect comments containing only 1 word.
F.e this post(https://www.instagram.com/p/Bsxlff2l9V8/) has 271 comment at the moment, BUT I can get only 253 comments.

Here is my code sample:

        // Firstly I collect them by minId
        $minId = null;
        do {
            $comments = $cl->media->getComments('1959512205109482876', ['min_id' => $minId]);

            foreach ($comments->getComments() as $comment) {
                $res[] = [
                    'username' => $comment->getUser()->getUsername(),
                    'user_id'  => $comment->getUser()->getPk(),
                    'time'     => $comment->getCreatedAt(),
                    'text'     => $comment->getText()
                ];
            }
            $minId = $comments->getNextMinId();
        } while ($minId !== null);

        // Than by maxId
        $maxId = null;
        do {
            $comments = $cl->media->getComments('1959512205109482876', ['max_id' => $maxId]);
            foreach ($comments->getComments() as $comment) {
                // Собираем все комменты в массив, чтобы после отсортировать и фильровать
                $res[] = [
                    'username' => $comment->getUser()->getUsername(),
                    'user_id'  => $comment->getUser()->getPk(),
                    'time'     => $comment->getCreatedAt(),
                    'text'     => $comment->getText()
                ];
            }
            $maxId = $comments->getNextMaxId();
        } while ($maxId !== null);

**Comment by JRoy** [comment]
**Please** use the issue template, next time you don't we'll close your issue without helping.

Additionally, please(!) post your debug logs.

**Comment by AndManowar** [comment]
> **Please** use the issue template, next time you don't we'll close your issue without helping.
> 
> Additionally, please(!) post your debug logs.
> 
> Also have you tested this on posts with smaller amounts of comments and does it work?

Yes I have. When the amount of comments smaller  - Im getting all comments.

**Comment by pashmak73** [comment]
Note that some of comments are replies! you must check for replies too!

**Comment by AndManowar** [comment]
> Note that some of comments are replies! you must check for replies too!

No, i did one-word comment and wasn`t able to find it in the comments list via API

**Comment by 27RU57** [comment]
Is it visible via regular app or website? Comment ban?

**Comment by AndManowar** [comment]
Yep, it is visible from my account in mobile app, but not from other account

**Comment by darkterminal** [comment]
**[CLOSED]**

**Comment by pilevar** [comment]
I have the same problem too.

Instagram app by default show top comments and users can switch between "Top Comments" and "Newest First", I think the problem is probably from this change.

**Comment by bolds07** [comment]
@pilevar , I think you are referring to an iphone feature of instagram app, I'm beta tester of instagram for android and i've no option to switch the coments sort criterea

it is common when listing followers or comments from some of them are not returned and some are duplicated, i would say until 2% of misses means an internal instagram server error and you should ignore... more than that is your fault
